File: moonpoison/binary_search_tree/700.cpp
#include<bits/stdc++.h>
using namespace std;
struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};
class Solution {
public:
    TreeNode* ans;
    TreeNode* searchBST(TreeNode* root, int val) 
    {
        search(root, val);
        return ans;
    }
    void search(TreeNode* root, int val)
    {
        if(!root){}
        else if(root->val==val) ans=root;
        else if(root->val>val) searchBST(root->left, val);
        else searchBST(root->right, val);
    }
};

File: ice0208/binary-search-tree/700.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def searchBST(self, root: TreeNode, val: int) -> TreeNode:
        cursor = root
        while cursor != None:
            if cursor.val == val:
                return cursor

            if val < cursor.val:
                cursor = cursor.left
            else:
                cursor = cursor.right

        return None
